### ISSUE TITLE
 [Web] [Version 3.10.0] Tappable area of `ElevatedButton` and `GestureDetector` is displaced, if it is inside `Stack` and `Align`

### ISSUE THREAD

**Issue by abigotado** [flutter/flutter#126946]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

In version 3.10.0 there is a problem un Flutter web with `ElevatedButton` when it is wrapped in `Align` widget inside `Stack`, which has `ListView` as the first child. The tappable area is displaced and doesn't coincide with rendered UI. The same problem occurs with `GestureDetector`. Without `Align` and `Stack` it sometimes occurs too (for example in Column, which has `ListView` as the first child).
The problem has appeared only after switching to Flutter 3.10.0 (it was a working code without any problems before upgrade), if I switch back to 3.7.12, buttons and gesture detectors work fine, as expected.

To reproduce - just create a `MaterialApp` with my piece of code (`Stack` and `Align` with `ElevatedButton` inside it).

### Expected results

Expected result: rendered `ElevatedButton` view coincides with it's tappable area. 

### Actual results

Actual result is up shift of the tappable area.

### Code sample

<details><summary>Code sample</summary>

```dart
Container(
        width: 399,
        color: Theme.of(context).colorScheme.surface,
        child: Stack(
          children: <Widget>[
                ListView(
                children: <Widget>[
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                  Text(AppLocalizations.of(context).wantParticipate),
                ],
              ),
            Align(
              alignment: Alignment.bottomCenter,
              child: Container(
                padding: const EdgeInsets.fromLTRB(24, 10, 24, 24),
                color: Theme.of(context).colorScheme.surface,
                child: ElevatedButton(
                  style: ElevatedButton.styleFrom(
                    maximumSize: const Size(double.maxFinite, 51),
                    minimumSize: const Size(double.maxFinite, 51),
                  ),
                  onPressed: () {},
                  child: Text(AppLocalizations.of(context).wantParticipate),
                ),
              ),
            ),
          ],
        ),
      ),
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

Tappable area is above the button.
<img width="359" alt="Снимок экрана 2023-05-16 в 16 25 20" src="https://github.com/flutter/flutter/assets/82471392/15f035e4-d3e9-4124-a70b-9f1b3b4380b1">

And most of the button is not clickable.
<img width="353" alt="Снимок экрана 2023-05-16 в 16 26 15" src="https://github.com/flutter/flutter/assets/82471392/5df500a4-4512-45a6-9571-416f43baf001">

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.0, on macOS 13.3.1 22E261 darwin-arm64, locale ru-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.1)
[✓] IntelliJ IDEA Community Edition (version 2022.2)
[✓] VS Code (version 1.78.0)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @abigotado 
Your sample code is not minimal so I have used below minimal sample code, but it works without issue:

<details>
<summary>Demo</summary>

https://github.com/flutter/flutter/assets/104349824/97926361-e419-4bc7-ac0e-e1dcc10c6860
</details>

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: const MyApp(),
    ),
  );
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Container(
          width: 399,
          color: Theme.of(context).colorScheme.surface,
          child: Stack(
            children: <Widget>[
              ListView(
                children: List<Widget>.generate(10, (index) => Text('Item $index'),)
              ),
              Align(
                alignment: Alignment.bottomCenter,
                child: Container(
                  padding: const EdgeInsets.fromLTRB(24, 10, 24, 24),
                  color: Theme.of(context).colorScheme.surface,
                  child: ElevatedButton(
                    style: ElevatedButton.styleFrom(
                      maximumSize: const Size(double.maxFinite, 51),
                      minimumSize: const Size(double.maxFinite, 51),
                    ),
                    onPressed: () {},
                    child: Text('Click here'),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```
</details>

Please try this sample code and confirm if it also works.

**Comment by abigotado** [comment]
@huycozy thank you very much for your answer! After some more research I found out that I had forgot about one thing - my app was wrapped in `ResponsiveWrapper` from [responisve_framework](https://pub.dev/packages/responsive_framework) package. If I remove `ResponsiveWrapper`, problem disappears. 

So that it seems that the problem is on on the side of the package. And there is an open issue here: #https://github.com/Codelessly/ResponsiveFramework/issues/140

**Comment by abigotado** [comment]
@huycozy Well, finally I've found out the where the problem lays - it is a `FittedBox` issue [responsive_framework](https://pub.dev/packages/responsive_framework) package has `FittedBox` as a base. And if I replace a `ResponsiveWrapper` widget with `FittedBox` the issue persists.

And I've found there is a discussion about this problem with fitted box:
#https://github.com/flutter/flutter/issues/126754

**Comment by ditman** [comment]
Related: https://github.com/flutter/flutter/issues/117091

**Comment by ditman** [comment]
@abigotado this PR might fix this issue: https://github.com/flutter/engine/pull/41870

**Comment by huycozy** [comment]
@abigotado Thanks for your confirmation. I will close this issue in favor of https://github.com/flutter/flutter/issues/126754 and to centralize on it. which may be fixed by PR David mentioned above. Please follow it for further updates.

**Comment by ditman** [comment]
The fix is coming to Flutter `master` soon: https://github.com/flutter/flutter/pull/127233 (we're also attempting to cherry-pick so it becomes available to everybody on `stable` (3.10)
